### PR TITLE
NIFI-11493: Fix properties dynamically modifying the classpath with default values

### PR DESCRIPTION
**Pull request by Lehel44**
# Summary

- Fixes PropertyDescriptors that dynamically modifies the classpath to work with default values
- Added new system test

[NIFI-11493](https://issues.apache.org/jira/browse/NIFI-11493)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/processor/DefaultedDynamicClassPathModificationIT.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        createFlow();
+
+        // Update modify to have the appropriate URL, don't update URL to load to let it on default value
+        final Map<String, String> propertyMap = new HashMap<>();
+        propertyMap.put("Class to Load", "org.apache.commons.lang3.StringUtils");
+        getClientUtil().updateProcessorProperties(defaultedModifyClasspathProcessor, propertyMap);
+        getClientUtil().waitForValidProcessor(defaultedModifyClasspathProcessor.getId());
+
+        // Create a FlowFile
+        getClientUtil().startProcessor(generateFlowFileProcessor);
```
Comment (markap14) [inline nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/processor/DefaultedDynamicClassPathModificationIT.java:52]: Need to waitForValidProcessor before starting it

File: nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/processor/DefaultedDynamicClassPathModificationIT.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        getClientUtil().stopProcessor(generateFlowFileProcessor);
+        getClientUtil().waitForStoppedProcessor(generateFlowFileProcessor.getId());
+
+        // Restart and ensure that everything works as expected after restart
+        getNiFiInstance().stop();
+        getNiFiInstance().start(true);
+
+        // Feed another FlowFile through. Upon restart, in order to modify, we need to get the most up-to-date revision so will first fetch the Processor
+        final ProcessorEntity generateAfterRestart = getNifiClient().getProcessorClient().getProcessor(generateFlowFileProcessor.getId());
+        getClientUtil().startProcessor(generateAfterRestart);
```
Comment (markap14) [inline nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/processor/DefaultedDynamicClassPathModificationIT.java:68]: Upon restart will need to wait for processor to become valid before starting

File: nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/processor/DefaultedDynamicClassPathModificationIT.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    private ConnectionEntity failureConnection;
+
+    @Test
+    void testLoadsClassFromDefaultedDynamicModification() throws NiFiClientException, IOException, InterruptedException {
+        createFlow();
+
+        // Update modify to have the appropriate URL, don't update URL to load to let it on default value
+        final Map<String, String> propertyMap = new HashMap<>();
+        propertyMap.put("Class to Load", "org.apache.commons.lang3.StringUtils");
+        getClientUtil().updateProcessorProperties(defaultedModifyClasspathProcessor, propertyMap);
```
[the remark this answers is not in the document]
Comment (Lehel44) [reply nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/processor/DefaultedDynamicClassPathModificationIT.java:46]: We should not set the 'URL to Load' property.

**Review by turcsanyip — APPROVED** [review]
Thanks for your review @markap14!

I also tested it with a simple flow using DBCPConnectionPool (locally modified to have a default value for the driver).

+1 LGTM
Merging to main and 1.x.

**Pull request closed without merging**